src/FluentContracts/Contracts/BaseContract.cs

```diff
@@ -63,5 +63,29 @@ public Linker<TContract> Satisfy<T, TException>(Func<T, bool> customCondition, s
         Validator.CheckGenericCondition<T, TException>(customCondition, (T)ArgumentValue, message);
         return _linker;
     }
+
+    public Linker<TContract> HasProperty(string propertyName, string? message = null)
```

Comment:
You should rename all the methods to start with `Have*`. Because it would not be grammatically correct to have `object.Must().HasProperty(...)`.